draco/core/io.py

```diff
@@ -713,42 +690,69 @@ class Truncate(task.SingleTask):
         Relative precision to truncate to (default 1e-4).
     variance_increase : float
         Maximum fractional increase in variance from numerical truncation.
+    ensure_chunked : bool
+        If True, ensure datasets are chunked according to their dataset_spec.
     """
 
     dataset = config.Property(proptype=list, default=None)
     weight_dataset = config.Property(proptype=list, default=None)
     fixed_precision = config.Property(proptype=float, default=None)
     variance_increase = config.Property(proptype=float, default=None)
+    ensure_chunked = config.Property(proptype=bool, default=True)
+
+    default_params = {
+        "weight_dataset": None,
+        "fixed_precision": 1e-4,
+        "variance_increase": 1e-3,
+    }
+
+    @functools.lru_cache()
```

Comment:
I think what's done in here is sufficiently quick it doesn't really warrant the LRU cache, but note that if we change this to a per instance call, then it'll never see the same pair twice and it should definitely be removed.

Reply:
Changes this to be from instance, and remove the lru_cache.